selfstat/stat.go

```
@@ -40,3 +40,7 @@ func (s *stat) Tags() map[string]string {
 	}
 	return m
 }
+
+func (s *stat) Unregister() {
```

Just to clarify the behaviour here: `stat` and `timingStat` are the two implementations of the Stat interface : `stat` is a simple counter/gauge, while `timingStat` calculates averages of timing samples.

The reason this use-after-unregister scenario is possible:
```go
stat := selfstat.Register("test", "field", tags)
stat.Incr(100)        // Works fine
stat.Unregister()     // Removes from registry only
stat.Incr(50)         // Still works locally, but data is "lost"
value := stat.Get()   // Returns 150, but won't appear in Metrics()
```
is because Unregister() only removes the stat from the global registry; the stat object itself continues to exist and function. This means the object keeps tracking values locally, but since it's no longer in the registry, those values won't be collected by Metrics().

This seems like intentional graceful degradation rather than failing hard, which makes sense for Telegraf's plugin lifecycle patterns.

---

I have added the suggestion, please check.